packages/experimental/Switch/src/useSwitch.ts

```diff
@@ -37,62 +31,27 @@ export const useSwitch = (props: SwitchProps, animationConfig?: AnimationConfig)
     ...rest
   } = props;
 
-  const onChangeWithAnimation = React.useCallback(
-    (e: InteractionEvent, checked?: boolean) => {
-      onChange && onChange(e, checked);
-      if (isMobile) {
-        if (checked) {
-          startTrackBackgroundAnimation(checked, trackBackgroundAnimation);
-          startTrackAnimation(false, animationConfig, animation, checked);
-        } else {
-          startTrackBackgroundAnimation(checked, trackBackgroundAnimation);
-          startTrackAnimation(false, animationConfig, animation, checked);
-        }
-      }
-    },
-    [onChange, trackBackgroundAnimation, animationConfig, animation],
-  );
-
-  const [checkedState, toggleCallback] = useAsToggleWithEvent(defaultChecked, checked, onChangeWithAnimation);
+  const [checkedState, toggleCallback] = useAsToggleWithEvent(defaultChecked, checked, onChange);
 
   //Setting the initial position of the knob on track when page loads.
   React.useEffect(() => {
     if (isMobile) {
-      if (checkedState) {
-        startTrackAnimation(false, animationConfig, animation, checkedState);
-        startTrackBackgroundAnimation(true, trackBackgroundAnimation);
-      } else {
-        startTrackAnimation(true, animationConfig, animation, checkedState);
-        startTrackBackgroundAnimation(false, trackBackgroundAnimation);
-      }
-    }
-  }, [checkedState, animationConfig]);
-
-  // Function to change background slowly over time when switch is toggled.
-  const startTrackBackgroundAnimation = React.useCallback((checked: boolean, animation: Animated.Value) => {
-    const toValue = checked ? 0 : 1;
-    Animated.timing(animation, {
-      toValue,
-      duration: 500,
-      useNativeDriver: false,
-    }).start();
-  }, []);
-
-  // Function to transform the knob over track in animated way when switch is toggled.
-  const startTrackAnimation = React.useCallback(
-    (onInit = false, animationConfig: AnimationConfig, animation: Animated.Value, toggled: boolean) => {
       Animated.timing(animation, {
-        toValue: toggled
-          ? animationConfig.trackWidth - (animationConfig.thumbWidth + animationConfig.thumbMargin * 2)
-          : onInit
-          ? 0
-          : -(animationConfig.trackWidth + animationConfig.thumbWidth),
-        duration: 300,
+        toValue: checkedState ? animationConfig.trackWidth - (animationConfig.thumbWidth + animationConfig.thumbMargin * 2) : 0,
+        duration: isInit ? 0 : 300,
         useNativeDriver: false,
       }).start();
-    },
-    [],
-  );
+
+      const toValue = checkedState ? 0 : 1;
+      Animated.timing(trackBackgroundAnimation, {
+        toValue,
+        duration: isInit ? 0 : 500,
+        useNativeDriver: false,
+      }).start();
+
+      setIsInit(false);
+    }
+  }, [checked, checkedState]);
```

Comment:
nit: doesn't look like this depends on checked anymore?

Reply:
@rurikoaraki , I tried with it but we have removed function with all the animation code in it and will be handling with this function itself.So when also checked state chnages we will need call this.